### PR TITLE
ready with task1 

### PR DESCRIPTION
**Pull request by Zornica**
add task1 - finished
add task2 - not finished
add task3 - finished

### CODE REVIEW

**Comment by mgenov** [comment]
When using maven, idea project files are useless, cause anyone is able to import project on it's own from the maven structure. You can add .idea directory to .gitignore too.

**Comment by Zornica** [comment]
without localhost:8080 in action the servlet doesn't start

**Comment by mgenov** [comment]
There is nothing in common between rendered HTML which is interpreted by the Browser and the real servlet configuration.

I saw several HTML pages which are placed in the package tree, where they should stay in the webapp folder. 

You can take a look at: https://docs.oracle.com/javaee/5/tutorial/doc/bnadx.html, there is a good picture that explains the correct WAR structure.

**Comment by mgenov** [comment]
Also tomcat has examples folder which several example apps. You can take a look into them too.

**Comment by Zornica** [comment]
Is that right or is not what you mean?

**Comment by mgenov** [comment]
Yes, I suppose it should work now, or at least the "ThreePagesServlet" should receive HTTP requests.

**Comment by Zornica** [comment]
Yes, it works.

**Comment by mgenov** [comment]
You only have to think for better name of the controller (ThreePagesServlet) cause it doesn't sound good, and to clean some of the not used code. I

**Comment by Zornica** [comment]
Are "ResponseServlet" or "MainServlet" good names?

**Comment by mgenov** [comment]
What is the role of this servlet?

**Comment by Zornica** [comment]
Recognizes from which page receives a request to transmit her name to another page to visualize it .

**Comment by mgenov** [comment]
You have to find a name that fits well to this description, and I would like from you to find out good name for it.

**Comment by Zornica** [comment]
I rename the servlet.

File: src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    <display-name>Archetype Created Web Application</display-name>
+
+    <servlet>
+        <servlet-name>servlet</servlet-name>
+        <servlet-class>com.clouway.servlets.task1.ThreeLinksServlet</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>servlet</servlet-name>
+        <url-pattern>/test</url-pattern>
```
Comment (mgenov) [inline src/main/webapp/WEB-INF/web.xml]: URL of recognition servlet is pointing to "/test" which is ambiguous.

File: src/main/webapp/WEB-INF/web.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    <servlet-mapping>
+        <servlet-name>pageThreeServlet</servlet-name>
+        <url-pattern>/pagethree</url-pattern>
+    </servlet-mapping>
+
+
+    <servlet>
+        <servlet-name>servlet2</servlet-name>
+        <servlet-class>com.clouway.servlets.task3.MyServlet</servlet-class>
```
Comment (mgenov) [inline src/main/webapp/WEB-INF/web.xml]: What is the role of MyServlet?

**Comment by mgenov** [comment]
Note that project files are still in this PR, and need to be cleared.

**Comment by Zornica** [comment]
I think I'm ready with fixes and task4

**Comment by mgenov** [comment]
Project files are still in there. 

http://stackoverflow.com/questions/1274057/making-git-forget-about-a-file-that-was-tracked-but-is-now-in-gitignore

**Comment by Zornica** [comment]
I removed .idea .

File: src/main/java/com/clouway/servlets/task3/CounterServlet.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ */
+public class CounterServlet extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("text/html");
+        HttpSession session = req.getSession();
+        String link = req.getParameter("link");
+        Integer count = (Integer) session.getAttribute(link);
+        PrintWriter out = resp.getWriter();
+        if (link != null) {
```
Comment (mgenov) [inline src/main/java/com/clouway/servlets/task3/CounterServlet.java]: Try to simplify this if statements. Embedded If statement is a code smell which need to be improved.

**Comment by mgenov** [comment]
I saw some things from Task4 ( I'm not sure that you need cookies), so you can push your changes, and create following PR for the next ones.

**Comment by Zornica** [comment]
I made some fixes in CounterServlet .

**Comment by Zornica** [comment]
I removed cookies and used ServletContext and RequestDispatcher. Is this good?

**Comment by mgenov** [comment]
You can push this to upstream, and make your changes in following PR, so it will be more easily for review.

**Pull request merged**